typescript/sdk/src/deploy/core/HyperlaneCoreDeployer.ts

```diff
@@ -184,38 +185,23 @@ export class HyperlaneCoreDeployer<
     };
   }
 
-  static async transferOwnership<CoreChains extends ChainName>(
-    core: HyperlaneCore<CoreChains>,
-    owners: ChainMap<CoreChains, types.Address>,
-    multiProvider: MultiProvider<CoreChains>,
-  ): Promise<ChainMap<CoreChains, ethers.ContractReceipt[]>> {
-    return promiseObjAll(
-      objMap(core.contractsMap, async (chain, coreContracts) =>
-        HyperlaneCoreDeployer.transferOwnershipOfChain(
-          coreContracts,
-          owners[chain],
-          multiProvider.getChainConnection(chain),
-        ),
-      ),
-    );
-  }
-
-  static async transferOwnershipOfChain(
-    coreContracts: CoreContracts,
-    owner: types.Address,
-    chainConnection: ChainConnection,
+  async transferOwnershipOfContracts(
+    chain: Chain,
+    ownables: Ownable[],
   ): Promise<ethers.ContractReceipt[]> {
-    // Mailbox ownership is transferred upon initialization.
-    const ownables: Ownable[] = [
-      coreContracts.multisigIsm,
-      coreContracts.proxyAdmin,
-    ];
-    return Promise.all(
-      ownables.map((ownable) =>
-        chainConnection.handleTx(
-          ownable.transferOwnership(owner, chainConnection.overrides),
-        ),
-      ),
+    const owner = this.configMap[chain].owner;
+    const chainConnection = this.multiProvider.getChainConnection(chain);
+    const receipts = await Promise.all(
+      ownables.map(async (ownable) => {
+        const currentOwner = await ownable.owner();
+        if (currentOwner.toLowerCase() !== owner.toLowerCase()) {
+          return chainConnection.handleTx(
+            ownable.transferOwnership(owner, chainConnection.overrides),
+          );
```

Comment:
as we add this idempotency everywhere, checker is increasingly less useful

Reply:
Totally. Though one thing checker does that the deployer does not (yet) is push txs to gnosis. So we can't deprecate it quite yet (though we probably should at some point)